client/client.cpp

```diff
@@ -18,9 +19,30 @@
 
 namespace {
 
+std::string makeUid(size_t len = 6) {
+  static std::mt19937* rng = nullptr;
+  // Initialize random number generator
+  if (rng == nullptr) {
+    std::random_device r;
+    std::seed_seq seed{r(), r(), r(), r(), r(), r(), r(), r()};
+    rng = new std::mt19937(seed);
+  }
+
+  static const char alphanum[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+  std::string s(len, 0);
+  for (size_t i = 0; i < len; i++) {
+    s[i] = alphanum[(*rng)() % (sizeof(alphanum) - 1)];
```

Comment:
I would have written something like:

```
std::uniform_int_distribution<int> index_distribution(0, sizeof(alphanum) - 1);
for (size_t i = 0; i < len; ++i) {
  int index_sample = index_distribution(*rng);
  s[i] = alphanum[index_sample];
}
```

It is a little bit more explicit in terms of operation semantics. But it's up to you, the presence of bias is not significant in UID generation.